cpp/src/fil/fil.cu

```diff
@@ -72,10 +72,10 @@ cat_sets_owner::cat_sets_owner(const std::vector<cat_feature_counters>& cf)
 
     int fid = max_matching.size();
```

Comment:
I'd slightly recommend using the following and leaving the format strings alone, since that avoids the implicit type conversion, but I'm not picky on that.
```suggestion
    auto fid = max_matching.size();
```